.github/workflows/pypi-release.yml

```diff
@@ -1,8 +1,9 @@
 name: Publish to PyPI
 on:
   push:
-    tags:
-      - "*"
+    branches: [main]  # this will just build and check package
```

Comment:
```suggestion
    branches: [main]  # this will just build and check package
    pull_request: [main]  # this will just build and check package
```